src/iceqube/queue.py

```diff
@@ -0,0 +1,111 @@
+from iceqube.classes import Job
+from iceqube.classes import State
+from iceqube.storage import Storage
+
+DEFAULT_QUEUE = "ICEQUBE_DEFAULT_QUEUE"
+
+
+class Queue(object):
+    def __init__(self, queue=DEFAULT_QUEUE, storage_path=None):
```

Comment:
So if you're making it resemble the RQ API, then I think you would want to accept a storage instance instead of the `storage_path`? I see `Queue` accepts an instance in the [example](https://github.com/rq/rq-scheduler#scheduling-a-job).

That pattern, a dependency injection pattern, you can replicate for most of the classes here, if need be. Some pros:
- It makes testing easier and the test coverage better because you're restricting the testing to the units you're focused on testing, while mocking the rest. For example, avoid covering `Storage` in tests for `Queue`
- Allows others to build their own storage implementation outside of this package
- Clearly separates concerns; a queue shouldn't really care how to build a storage instance, just that it needs one

The factory pattern also goes along with this, because you can add helper factories that will instantiate classes for you, helping recoup some of the inconvenience of building a Queue without just a `storage_path`. Like `build_queue_for_

Reply:
Yeah - I deviated deliberately from the RQ API in this respect in order to simplify the iceqube implementation.

We have not seen particular interest in iceqube, and as far as I am aware it is not being used anywhere except in Kolibri (it is definitely not being used anywhere else in Learning Equality).

I think it may be best to vendor it into Kolibri in future, hence the move towards a specific implementation and away from abstraction.

Reply:
Parallel work has brought this into much closer RQ API parity, so hopefully this is now addressed.